bin/tasks/snapshot.js

```diff
@@ -29,6 +33,82 @@ const TesterBuildQuery = `
   }
 `;
 
+const ReportQuery = `
+  query ReportQuery($buildNumber: Int!) {
+    app {
+      build(number: $buildNumber) {
+        createdAt
+        completedAt
+        snapshots {
+          status
+          spec {
+            name
+            component {
+              name
+              displayName
+            }
+          }
+          parameters {
+            viewport
+            viewportIsDefault
+          }
+        }
+      }
+    }
+  }
+`;
+
+export const generateReport = async (ctx, task) => {
+  const { client, log, options } = ctx;
+  const { number: buildNumber, reportToken } = ctx.build;
+
+  const file = options.report === true ? 'chromatic-test-report.xml' : options.report;
+  ctx.reportPath = path.resolve(file.replace(/{buildNumber}/g, buildNumber));
+
+  task.output = `Generating XML report at ${ctx.reportPath}`;
+
+  const { app } = await client.runQuery(
+    ReportQuery,
+    { buildNumber },
+    { Authorization: `Bearer ${reportToken}` }
+  );
+  const { createdAt, completedAt, snapshots } = app.build;
+  const buildTime = (completedAt || Date.now()) - createdAt;
+
+  const suite = reportBuilder
+    .testSuite()
+    .name(`Chromatic build ${buildNumber}`)
+    .time(Math.round(buildTime / 1000))
+    .timestamp(new Date(createdAt).toISOString());
+
+  snapshots.forEach(({ status, spec, parameters }) => {
+    const suffix = parameters.viewportIsDefault ? '' : ` [${parameters.viewport}px]`;
+    const testCase = suite
+      .testCase()
+      .className(spec.component.name)
+      .name(`${spec.name}${suffix}`);
+
+    switch (status) {
+      case 'SNAPSHOT_ERROR':
+        testCase.error('Server error while taking snapshot, please try again', status);
+        break;
+      case 'SNAPSHOT_CAPTURE_ERROR':
+        testCase.error('Snapshot is broken due to a JavaScript error in your Storybook', status);
+        break;
+      case 'SNAPSHOT_DENIED':
+        testCase.failure('Snapshot was denied by a user', status);
+        break;
+      case 'SNAPSHOT_NO_CAPTURE':
+        testCase.skipped();
+        break;
+      default:
```

Comment:
What about pending? These are the snapshots the user likely cares about

Reply:
Why yeah, pending should be a failure too of course.

Reply:
By the way the switch is only to set the error/failure/skipped flag, the pending snapshots were part of the report before, but marked 'successful'.